src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -0,0 +1,527 @@
+            # corrected flavours defined in BTA
+            Jet["flavour"] = ak.where(
+                jet.hadronFlavour != 0,
+                jet.hadronFlavour,
+                ak.where(
+                    (abs(jet.partonFlavour) == 4) | (abs(jet.partonFlavour) == 5),
+                    zeros,
+                    jet.partonFlavour,
+                ),
+            )
+
+            # genJet pT
+            genJetIdx = jet.genJetIdx.mask[
+                (jet.genJetIdx >= ak.num(events.GenJet)) & (jet.genJetIdx < 0)
+            ]  # in case the genJet index out of range
+            Jet["genpt"] = ak.fill_none(events.GenJet[genJetIdx].pt, -1)
+
+            # gen-level jet cleaning aginst prompt leptons
+            genlep_prompt = genlep[(Genlep.mother != 0) & (Genlep.mother % 10 == 0)]
+            overlapped = ak.zeros_like(jet.pt, dtype=bool)
+
+            for lep_pid, dr_thres in [(11, 0.2), (13, 0.2), (15, 0.3)]:
+                lep = genlep_prompt[abs(genlep_prompt.pdgId) == lep_pid]
+                pair = ak.cartesian(
+                    [jet, lep], axis=1, nested=True
+                )  # dim: (event, jet, lep)
+                dr_min = ak.min(
+                    pair["0"].delta_r(pair["1"]), axis=-1
+                )  # dim: (event, jet)
+                dr_min = ak.fill_none(dr_min, 99.0)  # for 0 lepton case, assign dr=99
+                overlapped = overlapped | (dr_min < dr_thres)
+            Jet["flavourCleaned"] = ak.where(
+                overlapped, zeros - 999, Jet["flavour"]
+            )  # fill -999 for jets with lepton overlapping
+
+            # jet cleaning aginst pileup
+            Jet["flavourCleaned"] = ak.where(
+                Jet["genpt"] < 8.0, zeros, Jet["flavourCleaned"]
+            )
+
+        # define Jet_clean: special for BTA ttbar workflow which cleans jets against selected leptons
+        overlapped = ak.zeros_like(jet.pt, dtype=bool)
+        for lep in [eles, muons]:
+            pair = ak.cartesian(
+                [jet, lep], axis=1, nested=True
+            )  # dim: (event, jet, lep)
+            dr_min = ak.min(pair["0"].delta_r(pair["1"]), axis=-1)  # dim: (event, jet)
+            dr_min = ak.fill_none(dr_min, 99.0)  # for 0 lepton case, assign dr=99
+            overlapped = overlapped | (dr_min < 0.4)
+
+        Jet_clean = Jet[overlapped == False]  # remove jets overlapping with leptons
+
+        ###############
+        #     MET     #
+        ###############
+        basic_vars["ttbar_met_pt"] = events.MET.pt
+        basic_vars["ttbar_met_phi"] = events.MET.phi
+
+        ###############
+        #  Selection  #
+        ###############
+        # apply selections on events based on BTA TTbarSelectionProducer
+        passLepSel = chsel != 0
+        passJetSel = (
+            ((abs(chsel) == 11) | (abs(chsel) == 13)) & (ak.num(Jet_clean) >= 4)
+        ) | ((abs(chsel) > 13) & (ak.num(Jet_clean) >= 1))
+        passMetSel = events.ChsMET.pt > 0
```

Comment:
Apologize  not noticing earlier the `MET` branch still takes `PFMET` as default collections.. should be `PuppiMET` for the MET variables
https://github.com/cms-sw/cmssw/blob/CMSSW_12_6_X/PhysicsTools/NanoAOD/python/met_cff.py#L49

Reply:
But only in case of Run 3

Reply:
Thanks for pointing this out!!! this is introduced when adding systematics, do it in next PR